openhands/core/config/README.md

```diff
@@ -0,0 +1,110 @@
+# Configuration Management in OpenHands
+
+## Overview
+
+OpenHands uses a flexible configuration system that allows settings to be defined through environment variables, TOML files, and command-line arguments. The configuration is managed through a package structure in `openhands/core/config/`.
+
+## Configuration Classes
+
+The main configuration classes are:
+
+- `AppConfig`: The root configuration class
+- `LLMConfig`: Configuration for the Language Model
+- `AgentConfig`: Configuration for the agent
+- `SandboxConfig`: Configuration for the sandbox environment
+- `SecurityConfig`: Configuration for security settings
+
+These classes are defined as dataclasses, allowing for easy type checking and default value specification.
+
+## Loading Configuration from Environment Variables
+
+The `load_from_env` function in the config package is responsible for loading configuration values from environment variables. It recursively processes the configuration classes, mapping environment variable names to class attributes.
+
+### Naming Convention for Environment Variables
+
+- Prefix: Uppercase name of the configuration class followed by an underscore (e.g., `LLM_`, `AGENT_`)
+- Field Names: Uppercase, with underscores replaced by uppercase letters
+- Full Variable Name: Prefix + Field Name (e.g., `LLM_API_KEY`, `AGENT_MEMORY_ENABLED`)
+
+### Examples
+
+```bash
+export LLM_API_KEY='your_api_key_here'
+export LLM_MODEL='gpt-4'
+export AGENT_MEMORY_ENABLED='true'
+export SANDBOX_TIMEOUT='300'
+```
+
+## Type Handling
+
+The `load_from_env` function attempts to cast environment variable values to the types specified in the dataclasses. It handles:
+
+- Basic types (str, int, bool)
+- Optional types (e.g., `str | None`)
+- Nested dataclasses
+
+If type casting fails, an error is logged, and the default value is retained.
+
+## Default Values
+
+If an environment variable is not set, the default value specified in the dataclass is used.
+
+## Nested Configurations
+
+The `AppConfig` class contains nested configurations like `LLMConfig` and `AgentConfig`. The `load_from_env` function handles these by recursively processing nested dataclasses with updated prefixes.
+
+## Security Considerations
+
+Be cautious when setting sensitive information like API keys in environment variables. Ensure your environment is secure.
+
+Certainly. I'll add a paragraph to the "## Usage" section that highlights the recommended way to use `load_app_config`. Here's the updated section:
+
+## Usage
+
+To load configuration from environment variables, TOML files, and apply final tweaks:
+
+1. Set the required environment variables
+2. Create a `config.toml` file with your desired settings
```

Comment:
We now have a config.template.toml. It feels like recommendations should be to use it, rather than ignore and create from scratch.

Reply:
Will remove that enumeration completely